ipn/ipnlocal/drive.go

```diff
@@ -243,7 +243,7 @@ func (b *LocalBackend) driveSetSharesLocked(shares []*drive.Share) error {
 		},
 		DriveSharesSet: true,
 	})
-	return b.pm.setPrefsLocked(prefs.View())
+	return b.pm.setPrefsNoCheck(prefs.View())
```

Comment:
this is weird that it's calling an unexported API directly. cc @oxtoacart

Reply:
Agreed. Perhaps this is a result of a misleading original name? Neither `setPrefsLocked` nor `SetPrefs` is concurrency-safe, but the Locked suffix on the former kind of suggests otherwise.

I think we should change it to
```go
return b.pm.SetPrefs(prefs.View(), ipn.NetworkProfile{})
```
but maybe in its own #cleanup PR?

Reply:
Use of `setPrefsNoPermCheck` looks ok to me.